isis/src/base/objs/Cube/Cube.cpp

```diff
@@ -2133,6 +2134,96 @@ namespace Isis {
   }
 
 
+/**
+ * Returns the latitude and longitude range for the Cube. More accurate than the minimum and 
+ * maximum latitude and longitude from the mapping group. 
+ * 
+ * @param minLatitude minimum latitude of the cube
+ * @param maxLatitude maximum latitude present in the cube
+ * @param minLongitude minimum longitude present in the cube
+ * @param maxLongitude maximum longitude present in the cube
+ */
+  void Cube::latLonRange(double &minLatitude, double &maxLatitude, double &minLongitude, double &
+                         maxLongitude) {
+    Camera *cam;
+    TProjection *proj;
+
+    bool isGood = false;
+    bool noCamera;
+
+    if (hasGroup("Instrument")) {
+      noCamera = false;
+    }
+    else {
+      noCamera = true;
+    }
+
+    // setup camera or projection
+    if (noCamera) {
+     try {
+       proj = (TProjection *) projection();
+     }
+     catch(IException &e) {
+       QString msg = "Cannot calculate lat/lon range without a camera or projection";
+       throw IException(e, IException::User, msg, _FILEINFO_);
+     }
+    }
+    else {
+      try {
+        cam = camera();
+      }
+      catch(IException &e) {
+        QString msg = "Unable to create camera when calculating a lat/lon range.";
+        throw IException(e, IException::User, msg, _FILEINFO_);
+      }
+    }
+
+    // Iterate over all samp/line combos in cube
+    minLatitude = 99999;
+    minLongitude = 99999;
+    maxLatitude = -99999;
+    maxLongitude = -99999;
```

Comment:
If nothing in the image successfully projects, then these will get returned. I'm not sure how to deal with this case or even if we should at all.

Reply:
Good point.

Reply:
I updated it to throw an exception if any of these are still set to their original 9999 values.

Reply:
Updated this to throw an exception if any of these remain non-updated.